server/src/main/java/org/opensearch/search/builder/SearchSourceBuilder.java

```diff
@@ -347,6 +350,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalBoolean(seqNoAndPrimaryTerm);
         out.writeNamedWriteableList(extBuilders);
         out.writeBoolean(profile);
+        out.writeOptionalString(searchPipeline);
```

Comment:
You'll need to guard serialization and deserialization of this field with version checks.

Reply:
Was just about to push that :)